# Generated by Django 3.2.7 on 2021-11-08 10:43

from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ('prob', '0001_initial'),
    ]

    operations = [
        migrations.CreateModel(
            name='UserProfile',
            fields=[
                ('user', models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, primary_key=True, related_name='profile', serialize=False, to='prob.user')),
                ('introduction', models.TextField(default='')),
            ],
        ),
        migrations.CreateModel(
            name='ProblemSet',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('title', models.CharField(default='default title', max_length=100)),
                ('date', models.DateTimeField(auto_now_add=True)),
                ('type', models.BooleanField(default=False)),
                ('tag', models.CharField(default='default tag', max_length=100)),
                ('difficulty', models.SmallIntegerField(default=0)),
                ('content', models.TextField(default='default content', max_length=1000)),
                ('creator', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='created_problem', to='prob.userstatistics')),
                ('recommender', models.ManyToManyField(to='prob.UserStatistics')),
            ],
        ),
        migrations.AlterField(
            model_name='solved',
            name='problem',
            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='prob.problemset'),
        ),
        migrations.DeleteModel(
            name='Problem',
        ),
    ]
